target/linux/mvebu/base-files/lib/preinit/06_set_iface_mac

```diff
@@ -40,6 +40,14 @@ preinit_set_mac_address() {
 			;;
 		esac
 		;;
+	wd,mcex2u)
+		if [ -f /tmp/wd-config/mac_addr ]; then
+			mac=$(cat /tmp/wd-config/mac_addr)
```

Comment:
So, what's the condition of this file to be available?

Reply:
It is a configuration file from the vendor config partition being mounted in `preinit_mount_config`. It might be corrupted or removed, so the script checks if it is present before trying to fetch the MAC.

Reply:
okay, so there is a legitimate reason why the file wouldn't be there. I just want to make sure there is a reason behind these kind of checks...

However, don't think checking with -f won't help much if the file is corrupted :-)

Reply:
It is for the case when the config filesystem is corrupted and could not be mounted.